src/pages/GamePage.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useRef } from "react";
+import { StyledEngineProvider } from "@mui/material/styles";
+
+import { makeStyles } from "@mui/styles";
+import { Typography, Box } from "@mui/material";
+
+import formatTime from "../utils/utils";
+
+const useStyles = makeStyles(() => ({
+  statsContainer: {
+    display: "flex",
+    justifyContent: "center",
+  },
+  stats: {
+    display: "inline",
+    fontSize: "30px",
+    margin: "0 30px",
+  },
+  container: {
+    height: "100vh",
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  textBox: {
+    padding: "30px",
+    marginBottom: "30px",
+    maxWidth: "70%",
+  },
+  blueFont: {
+    color: "#2bbeed",
+    display: "inline",
+    fontSize: "50px",
+  },
+  redFont: {
+    backgroundColor: "#e0e0e0",
+    color: "red",
+    display: "inline",
+    fontSize: "50px",
+  },
+  greyFont: {
+    color: "grey",
+    display: "inline",
+    fontSize: "50px",
+  },
+  blackFont: {
+    backgroundColor: "#e0e0e0",
+    display: "inline",
+    fontSize: "50px",
+  },
+}));
+
+const GamePage = () => {
+  const classes = useStyles();
+
+  // temporary typing text
+  const typingText =
+    "import React, { useState, useEffect, useRef } from 'react'; const [isMissType, setIsMissType] = useState<boolean>(false);";
+
+  // const [typingText, setTypingText] = useState("");
+  const [currentIndex, setCurrentIndex] = useState<number>(0);
+  const [isMissType, setIsMissType] = useState<boolean>(false);
+  const [missCount, setMissCount] = useState<number>(0);
+  const [finished, setFinished] = useState<boolean>(false);
+  const [started, setStarted] = useState<boolean>(false);
+  const [timeTyping, setTimeTyping] = useState<number>(0);
+
+  const timer = useRef<ReturnType<typeof setInterval> | null>(null);
```

Comment:
```suggestion
  const timer = useRef<NodeJS.Timer | null>(null);
```
上記にように直接NodeJS.Timerを指定しても大丈夫そうだったので、よりシンプルにしましょうか！